packages/mermaid/src/diagrams/mindmap/mindmapDb.ts

```diff
@@ -23,34 +24,32 @@ const getParent = function (level) {
   return null;
 };
 
-export const getMindmap = () => {
+const getMindmap = () => {
   return nodes.length > 0 ? nodes[0] : null;
 };
-export const addNode = (level, id, descr, type) => {
+
+const addNode = (level: number, id: string, descr: string, type: number) => {
   log.info('addNode', level, id, descr, type);
   const conf = getConfig();
+  let padding: number = conf.mindmap.padding;
+  switch (type) {
+    case nodeType.ROUNDED_RECT:
+    case nodeType.RECT:
+    case nodeType.HEXAGON:
+      padding *= 2;
+  }
+
   const node = {
     id: cnt++,
-    nodeId: sanitizeText(id),
+    nodeId: sanitizeText(id, conf),
     level,
-    descr: sanitizeText(descr),
+    descr: sanitizeText(descr, conf),
     type,
     children: [],
-    width: getConfig().mindmap.maxNodeWidth,
-  };
-  switch (node.type) {
-    case nodeType.ROUNDED_RECT:
-      node.padding = 2 * conf.mindmap.padding;
-      break;
-    case nodeType.RECT:
-      node.padding = 2 * conf.mindmap.padding;
-      break;
-    case nodeType.HEXAGON:
-      node.padding = 2 * conf.mindmap.padding;
-      break;
-    default:
-      node.padding = conf.mindmap.padding;
-  }
+    width: conf.mindmap?.maxNodeWidth ?? 200,
```

Comment:
```suggestion
    width: conf.mindmap.maxNodeWidth,
```

Reply:
> Does it actually have all values defined? 
It _might_ not. But the type of defaultConfig is `RequiredDeep<MermaidConfig>`, so we're assuming it has all the values. 

> Does it handle the case when the user set an attribute as undefined?

No, runtime changes are not handled. 


Actually, I've removed that type now. We can reintroduce after some discussion in a separate PR.